Barotrauma/BarotraumaShared/SharedSource/Items/Components/Signal/GreaterComponent.cs

```diff
@@ -5,7 +5,8 @@ namespace Barotrauma.Items.Components
 {
     class GreaterComponent : EqualsComponent
     {
-        private float val1, val2;
+        private float val1 = float.NegativeInfinity;
+        private float val2 = float.PositiveInfinity;
```

Comment:
In my opinion the use of these sentinel values isn't the clearest way to go about this. I think it'd make the intention clearer if these were for example nullable floats, where null represents "no value received".